security/pkg/pki/ca/ca.go

```diff
@@ -148,17 +148,20 @@ func NewSelfSignedIstioCAOptions(ctx context.Context, caCertTTL, certTTL, maxCer
 			return nil, fmt.Errorf("failed to create CA KeyCertBundle (%v)", err)
 		}
 
-		// Rewrite the key/cert back to secret so they will be persistent when CA restarts.
+		// Write the key/cert back to secret so they will be persistent when CA restarts.
 		secret := BuildSecret("", CASecret, namespace, nil, nil, nil, pemCert, pemKey, istioCASecretType)
 		if _, err = client.Secrets(namespace).Create(secret); err != nil {
-			log.Errorf("Failed to write secret to CA (error: %s). This CA will not persist when restart.", err)
+			log.Errorf("Failed to write secret to CA (error: %s). Abort.", err)
+			return nil, fmt.Errorf("failed to create CA due to secret write error")
```

Comment:
What will happen here?  The Citadel process will exit?

Reply:
That's correct. The process will exit if the CA-root secret already exists and Citadel tries to overwrite it. In this case, the Citadel will be restarted by Kubelet, to read the CA-root secret again.

I think this is good and safe enough for this fix. We may implement retries in the future if necessary.